### ISSUE TITLE
Postfix numbers are shown in scientific notation 

### ISSUE THREAD

**Issue by HatScripts** [tqdm/tqdm#621]
- [x] I have visited the [source website], and in particular, read the [known issues]
- [x] I have searched through the [issue tracker] for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
```py
print(tqdm.__version__)  # 4.26.0
print(sys.version)  # 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 17:00:18) [MSC v.1900 64 bit (AMD64)]
print(sys.platform)  # win32
```

All numbers greater than 999 in the postfix are shown in the progress bar in scientific notation.
Examples:

Input | Output
-------|-------
`1000` | `1e+03`
`1005` | `1.00e+03`
`1174` | `1.17e+03`
`1509` | `1.51e+03`

As you can see (at least for smaller numbers) it's
* harder to read
* less precise
* more verbose

Is this the intended behaviour? If so, how can I disable it? The `unit_scale` and `unit_divisor` params in the constructor looked promising, but they don't seem to be related to this.

Here's a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example):
```py
from tqdm import tqdm

with tqdm() as bar:
    bar.set_postfix({"some_number": 1000})
```

which outputs:
```
0it [00:00, ?it/s, some_number=1e+03]
```

  [source website]: https://github.com/tqdm/tqdm/
  [known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
  [issue tracker]: https://github.com/tqdm/tqdm/issues?q=

**Comment by casperdcl** [comment]
you can always use e.g. `"{}".format(1000)` instead of `1000`

**Comment by HatScripts** [comment]
@casperdcl While I appreciate your response, I was more so hoping for an explanation for this behaviour, or an acknowledgement that it's a bug (or at least that it shouldn't be the default behaviour), rather than a workaround that would require me to manually wrap every int value I pass to `set_postfix` with `str()`.

I use tqdm quite a lot throughout my projects, and call `set_postfix` quite a lot, often with multiple integer values in the dict, e.g.
```py
bar.set_postfix({
    "dupe_groups": dupe_groups,
    "dupes"      : dupes,
    "dupes/group": dupes / dupe_groups if dupe_groups > 0 else "-",
    "file_size"       : file_size,
    "depth"       : depth,
    # etc.
})
```

It's just honestly not very intuitive that it formats every int in the postfix in scientific notation. It would be like if it formatted every int in the postfix as hexadecimal with the only way to prevent the behaviour being to convert my ints to strings beforehand.

**Comment by casperdcl** [comment]
It's mentioned in the documentation at https://github.com/tqdm/tqdm#description-and-additional-stats that there's this auto-conversion. You could also play around with `bar_format` to get what you want. I agree it's silly though to have `1001 = 1.00e+03` and happy to change that. Not sure about arbitrarily large integers though - it's a feature not a bug implemented by @lrq3000

**Comment by HatScripts** [comment]
Yes, I agree about arbitrary large numbers, or floats with arbitrarily many decimal places.

So for the best of both worlds, rather than the current [`'{0:2.3g}'.format(n)`](https://github.com/tqdm/tqdm/blob/master/tqdm/_tqdm.py#L1184), perhaps a function like the following could be used:
```py
def format_num(n):
    f = '{0:.3g}'.format(n)
    f = f.replace('+0', '+')
    f = f.replace('-0', '-')
    n = str(n)
    return f if len(f) < len(n) else n
```

It simply ensures that the formatted number will never exceed the length of the number itself.

n (Number) | `format_num(n)` (str) | `'{0:2.3g}'.format(n)` (str)
---|---|---
`1000`     | `1000`     | `1e+03`
`1005`     | `1005`     | `1.00e+03`
`1174`     | `1174`     | `1.17e+03`
`1509`     | `1509`     | `1.51e+03`
`100000`   | `1e+5`    | `1e+05`
`1000000`  | `1e+6`    | `1e+06`
`1247237`  | `1247237`  | `1.25e+06`
`12472370` | `1.25e+7` | `1.25e+07`
`0.00000`  | `0`        | `' 0'` <- note the space*
`0.55555`  | `0.556`    | `0.556`
`3.14159`  | `3.14`     | `3.14`
`0.00001` | `1e-5` | `1e-05`

\*It seems the 2 in `'{0:2.3g}'.format(n)` pads with spaces so that the formatted string is at least 2 chars long. I'm not really sure what the reason for this is, as it only affects numbers 0 through 9.
